pkg/kudoctl/packages/verifier/template/verify_extended_parameters.go

```diff
@@ -0,0 +1,138 @@
+package template
+
+import (
+	"strings"
+
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/verifier"
+)
+
+var _ packages.Verifier = &ExtendedParametersVerifier{}
+
+type ExtendedParametersVerifier struct {
+	VerifyParamDescription bool
+	VerifyParamHint        bool
+	VerifyParamDisplayName bool
+	VerifyParamType        bool
+	VerifyParamGroup       bool
+	VerifyGroups           bool
+}
+
+// Verify implements packages.Verifier for parameter verification
+func (v *ExtendedParametersVerifier) Verify(pf *packages.Files) verifier.Result {
+
+	res := verifier.NewResult()
+	if v.VerifyParamDescription {
+		res.Merge(v.verifyDescription(pf))
+	}
+	if v.VerifyParamHint {
+		res.Merge(v.verifyHint(pf))
+	}
+	if v.VerifyParamDisplayName {
+		res.Merge(v.verifyDisplayName(pf))
+	}
+	if v.VerifyParamGroup {
+		res.Merge(v.verifyParamGroup(pf))
+	}
+	if v.VerifyParamType {
+		res.Merge(v.verifyType(pf))
+	}
+	if v.VerifyGroups {
+		res.Merge(v.verifyGroups(pf))
+	}
+	return res
+}
+
+func (ExtendedParametersVerifier) verifyDescription(pf *packages.Files) verifier.Result {
+	res := verifier.NewResult()
+	for _, p := range pf.Params.Parameters {
+		if p.Description == "" {
+			res.AddParamWarning(p.Name, "has no description")
+		} else {
+			if !strings.HasSuffix(p.Description, ".") {
+				res.AddParamWarning(p.Name, "has a description not ending with a '.'")
+			}
+		}
+	}
+	return res
+}
+
+func (ExtendedParametersVerifier) verifyDisplayName(pf *packages.Files) verifier.Result {
+	res := verifier.NewResult()
+	for _, p := range pf.Params.Parameters {
+		if p.DisplayName == "" {
+			res.AddParamWarning(p.Name, "has no displayName")
+		}
+		if strings.HasSuffix(p.DisplayName, ":") {
+			res.AddParamWarning(p.Name, "has a displayName ending with ':'")
+		}
+	}
+	return res
+}
+
+func (ExtendedParametersVerifier) verifyHint(pf *packages.Files) verifier.Result {
+	res := verifier.NewResult()
+	for _, p := range pf.Params.Parameters {
+		if p.Hint == "" {
+			res.AddParamWarning(p.Name, "has no hint")
+		} else {
+			if !strings.HasSuffix(p.Hint, ".") {
+				res.AddParamWarning(p.Name, "has a hint not ending with a '.'")
+			}
+		}
+	}
+	return res
+}
+
+func (v ExtendedParametersVerifier) verifyParamGroup(pf *packages.Files) verifier.Result {
+	res := verifier.NewResult()
+	for _, p := range pf.Params.Parameters {
+		if p.Group == "" {
+			res.AddParamWarning(p.Name, "has no group")
```

Comment:
This would probably be too verbose - top-level parameters without a group are perfectly fine, aren't' they?

Reply:
Well... Maybe? :) Generally I agree, that's why this is a separate check, but in this case I wanted to know which parameters don't have a group assigned yet, thats why this check is here...